### PR TITLE
🎉 feat v0.1

### PR DESCRIPTION
**Pull request by jkcs**
Fixes skytable/skytable#324
/claim skytable/skytable#324

# TODO LIST
- [x]  Test the usability

### CODE REVIEW

**Comment by ohsayan** [comment]
Great work @jkcs! I'll review this in a few moments.

**Comment by jkcs** [comment]
@ohsayan Thank you for the help you provided on Discord. After testing and fixing it, everything is working perfectly now. As for the suggestion you made about refactoring it into bl, can we postpone it for now? (I have less free time available currently.)

**Comment by jkcs** [comment]
Just a side note, there are some errors and omissions in the content of the "Networking" section at https://docs.skytable.io/protocol/networking. It seems that Skytable also has a bug regarding the return of integer types. Once I  find the steps to reproduce it, I will raise an issue.

**Review by ohsayan — CHANGES_REQUESTED** [review]
Great work! Would be great if you can add some documentation comments.

File: src/Query.ts [hunk 1]
```diff
@@ -1,42 +0,0 @@
...
-    private param_cnt: number;
-
-    /**
-     * Create a new query with the base query set. You can now add (any) additional parameters.
-     * 
-     * @param query The base query
-     */
-    constructor(query: string) {
-        this.query_buffer = new Uint8Array(Buffer.from(query, 'utf-8'));
-        this.param_buffer = new Uint8Array();
```
Comment (ohsayan) [inline src/Query.ts:13]: We need a `Query` object! This will allow adding multiple parameters, say in a loop for example.

File: src/config.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    const data = await connectionWrite(connect, getClientHandshake(this));
+
+    await bufferToHandshakeResult(data);
+
+    this.connection = connect;
+
+    return createSkytable(connect);
+  }
+
+  async connectTSL(options: ConnectionTLSOptions) {
```
Comment (ohsayan) [inline src/config.ts]: Looks like this is a typo. It should be TLS

File: src/protocol.ts [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+          return [
+            param < 0 ? PARAMS_TYPE.SINT : PARAMS_TYPE.UINT,
+            String(param),
+            '\n',
+          ].join('');
+        case 'boolean':
+          return [PARAMS_TYPE.BOOLEAN, Number(param) === 1 ? '\x01' : 0].join(
+            '',
+          );
+        default:
```
Comment (ohsayan) [inline src/protocol.ts]: Is this safe to do? If there is a custom type we can't assume it to be a binary blob

File: src/skytable.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  | ColumnBinary
+  | ColumnList<ColumnBase>;
+
+export type Row = Column[];
+
+export type Rows = Row[];
+
+export type QueryResult = Column | Row | Rows;
+
+export function createSkytable(connection: Socket | TLSSocket) {
```
Comment (ohsayan) [inline src/skytable.ts]: If this function is exported, I think we'll want to call it something like `createConnection` or `connectDB`

**Review by ohsayan — CHANGES_REQUESTED** [review]

**Comment by jkcs** [comment]
@ohsayan The points you mentioned have all been updated

**Comment by jkcs** [comment]
@ohsayan The `buffer.subarray()` method returns a new Buffer object that shares memory with the original Buffer object. It allows you to access a portion of the original Buffer object without copying the memory.

I think it's unnecessary for us to refactor the performance improvements into `bl`. The `slice` method in Buffer is already deprecated, and using `subarray` is considered best practice.

File: src/config.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+/**
+ * Configuration for a client connection (single node)
+ */
+export class Config {
+  private username: string;
+  private password: string;
+  private host: string;
+  private port: number;
+  private connection: Socket | TLSSocket | undefined;
```
Comment (ohsayan) [inline src/config.ts:22]: Please `export default` the `Config` class here

**Review by ohsayan — CHANGES_REQUESTED** [review]
I attempted to import the package to test with this `package.json`:

```json
{
  "name": "jsclient-testbench",
  "version": "1.0.0",
  "main": "dist/index.ts",
  "author": "Sayan Nandan <nandansayan@outlook.com>",
  "license": "Apache-2.0",
  "devDependencies": {
    "typescript": "^5.3.3"
  },
  "dependencies": {
    "skytable": "github:jkcs/client-nodejs#feat/v0.1"
  },
  "scripts": {
    "build": "tsc",
    "start": "tsc && node dist/index.js"
  }
}
```

and with contents in `src/index.ts`:
```ts
import {Config} from 'skytable';

const cfg = new Config("root", "pass");
```

Are the exports set up correctly to be accessed like this? If not, what is the export structure. An integration test like example would really help!

File: src/config.ts (hunk 5 above)
Comment (jkcs) [reply src/config.ts:22]: We can make the changes in index.ts.

**Comment by jkcs** [comment]
@ohsayan 
## 1. How to publish?
1. run `yarn`
2. run `yarn build`
3. Publishing to a remote repository in npm.（`npm login`、`npm publish`）

## 2. How to use?
> You can use `npm link` for local testing.
> 1.  `/client-nodejs` dir, run `npm link`.
> 2. Other project, run `npm link skytable`
```js
import Config from 'skytable';
const config = new Config('root', 'admin');

await db.query(`create space IF NOT EXISTS spaceName`);
```

**Comment by jkcs** [comment]
@ohsayan I have tested it, and it is ready for publishing. You can change the version number in the package.json file for the initial publishing.

**Comment by jkcs** [comment]
```
"dependencies": {
    "skytable": "github:jkcs/client-nodejs#feat/v0.1"
  },
```
Testing cannot be done in this way because this is the source code and there are no build.

**Review by ohsayan — APPROVED** [review]
@jkcs thanks for your comment. We already have the publish script set up at [publish-npm.yml](https://github.com/skytable/client-nodejs/blob/next/.github/workflows/publish-npm.yml) so don't worry!

I completely forgot about the fact that with GH as the source it won't build.

**Comment by ohsayan** [comment]
⚠️ This does not handle unbuffered cases.

**Pull request merged**